### ISSUE TITLE
[Bug] missing xdg_foreign support

### ISSUE THREAD

**Issue by superiums** [AppFlowy-IO/AppFlowy#4114]
### Bug Description

when try to export markdown or change the data dir,
the dialog not display, report following err:
```bash

(AppFlowy:4539): Gdk-WARNING **: 11:21:23.973: Server is missing xdg_foreign support
  2023-12-07 11:33:07 ERROR  Uncaught platform error
    at dart-ffi/src/lib.rs:156
```

### How to Reproduce

i'm on void linux. and dwl and wayland.

just click the button for chane data dir, or click share markdown.

maybe some lib needed ? i don't know which one it is.

is it possible to use the commen way like other apps use (eg. chromium, gedit....) ? they could open the filechoose dialog correctlly.

### Expected Behavior

file choose dialog appears.

### Operating System

void-linux

### AppFlowy Version(s)

0.3.8

**Comment by sburris0** [comment]
Also seeing this on my machines:
* nixos + sway + wayland
* nixos + gnome + wayland
So linux/wayland is the common denominator here

Sounds like maybe a file selector plugin is needed? https://stackoverflow.com/questions/64620106/how-to-open-graphical-files-picker-in-flutter-on-linux

**Comment by LucasXu0** [comment]
@sburris0 Yes, we're using the file picker package for selecting a file. However, the documentation din't mention that there is a required library/package needed for installation. Therefore, I'm unsure how to fix it, as it works fine on my Ubuntu machine.
